addons/cssresources/src/css-resource-panel.tsx

```diff
@@ -17,12 +17,12 @@ interface CssResourcePanelProps {
   };
 }
 
-interface CssResourcePanelState {
+interface State {
   list: CssResource[];
 }
 
-export class CssResourcePanel extends Component<CssResourcePanelProps, CssResourcePanelState> {
-  constructor(props: CssResourcePanelProps) {
+export class CssResourcePanel extends Component<Props, State> {
```

Comment:
Maybe for step 2 but just for showing what's possible if you want to have it 100% accurate
```typescript
interface Props<TEvent, TData> {
  active: boolean;
  api: {
    emit(event: TEvent, data: TData): void;
  };
}

export class CssResourcePanel<TEvent = (...args) => any, TData = any> extends Component<Props<TEvent, TData> , State> {
}
```

Depending on the use case (which I don't know) devs could now make their CssResourcePanel stuff Type-safe if they want to. Default is `any`

Reply:
Yeah this gives me some idea on how to get this maybe: `Prop<API>`